Index: src/main/java/edu/sjsu/cs/Store_Properties/Store.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/edu/sjsu/cs/Store_Properties/Store.java b/src/main/java/edu/sjsu/cs/Store_Properties/Store.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/edu/sjsu/cs/Store_Properties/Store.java	
@@ -0,0 +1,39 @@
+package edu.sjsu.cs.Store_Properties;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.sjsu.cs.User.Product;
+
+
+
+public class Store {
+    private static Store instance = null;
+    private String name;
+    private Map<String, Product> products;
+
+    private Store(String name) {
+        this.name = name;
+        this.products = new HashMap<>();
+    }
+
+    public static Store getInstance(String name) {
+        if (instance == null) {
+            instance = new Store(name);
+        }
+        return instance;
+    }
+
+    public void addProduct(Product product) {
+        products.put(product.getProductId(), product);
+    }
+
+    public void removeProduct(Product product) {
+        products.remove(product.getProductId());
+    }
+
+    public Map<String, Product> getProducts() {
+        return products;
+    }
+}
\ No newline at end of file
Index: src/main/java/edu/sjsu/cs/User/Product.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/edu/sjsu/cs/User/Product.java b/src/main/java/edu/sjsu/cs/User/Product.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/edu/sjsu/cs/User/Product.java	
@@ -0,0 +1,79 @@
+package edu.sjsu.cs.User;
+
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+public class Product {
+    private String productId;
+    private Image image;
+    private String name;
+    private double price;
+    private int quantity;
+    private String description;
+    private double totalPrice;
+
+    public Product(String productId, String name, double price, String description) {
+        this.productId = productId;
+        this.name = name;
+        this.price = price;
+        this.description = description;
+        this.quantity = 1;
+        totalPrice = price * quantity;
+    }
+
+    public String getProductId() {
+        return productId;
+    }
+
+    public void setProductId(String productId) {
+        this.productId = productId;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public void setPrice(double price) {
+        this.price = price;
+    }
+
+    public int getQuantity(){
+        return quantity;
+    }
+
+    public void setQuantity(int quantity){
+        this.quantity = quantity;
+    }
+
+    public String getDescription(){
+        return description;
+    }
+
+    public void setDescription(String description){
+        this.description = description;
+    }
+
+    public Image getImage(){
+        return image;
+    }
+
+    public void setImage(Image image){
+        this.image = image;
+    }
+
+    public double getTotalPrice(){
+        return totalPrice;
+    }
+
+    public void setTotalPrice(double totalPrice){
+        this.totalPrice = totalPrice;
+    }
+}
\ No newline at end of file
Index: src/main/java/edu/sjsu/cs/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/edu/sjsu/cs/Main.java b/src/main/java/edu/sjsu/cs/Main.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/edu/sjsu/cs/Main.java	
@@ -0,0 +1,95 @@
+ppackage edu.sjsu.cs;
+
+
+import edu.sjsu.cs.User.Customer;
+import edu.sjsu.cs.User.Product;
+import edu.sjsu.cs.User.Seller;
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.stage.Stage;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+
+public class Main extends Application {
+    private static Store store = new Store("eMart");
+    private static Customer customer = new Customer ("customer", "password", "email", "address", "phoneNumber");
+    private static Seller seller = new Seller ("customer", "password", "email", "phoneNumber", "company", "address");
+    public static Seller getSeller(){return seller;}
+
+    public static Customer getCustomer(){
+        return customer;
+    }
+
+    public static Store getStore() {
+        return store;
+    }
+
+    @Override
+    public void start(Stage stage) {
+        //Make a customer
+        //Add items to cart
+        store.getProducts().put("1", new Product("1", "apple", 10.0));
+        store.getProducts().put("2", new Product("2", "banana", 15.0));
+        store.getProducts().put("3", new Product("3", "dragon fruit", 20.0));
+        store.getProducts().put("4", new Product("4", "orange", 25.0));
+
+        try {
+            Parent root = FXMLLoader.load(getClass().getResource("CustomerMainPage.fxml"));
+            Scene scene = new Scene(root);
+            stage.setScene(scene);
+            stage.show();
+
+
+
+        } catch(Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void main(String[] args) {
+        launch(args);
+    }
+}
+
+// import javafx.scene.layout.BackgroundFill;
+// import javafx.scene.layout.BorderPane;
+// import javafx.scene.layout.CornerRadii;
+// import javafx.geometry.Insets;
+// import javafx.geometry.Pos;
+// import javafx.application.Application;
+// import javafx.scene.Scene;
+// import javafx.scene.control.Label;
+// import javafx.scene.image.ImageView;
+// import javafx.scene.layout.AnchorPane;
+// import javafx.scene.layout.Background;
+// import javafx.scene.layout.FlowPane;
+// import javafx.scene.layout.Pane;
+// import javafx.scene.paint.Color;
+// import javafx.stage.Stage;
+
+// import java.util.ArrayList;
+// import java.util.List;
+
+// public class Main extends Application {
+
+//     @Override
+//     public void start(Stage stage) {
+//         List<AnchorPane> items = new ArrayList<>();
+//         items.add(new CustomAnchorPane());
+//         FlowPane flowPane = new FlowPane();
+//         flowPane.setPrefWidth(200);
+//         flowPane.setMaxWidth(200);
+
+//         for (AnchorPane item : items) {
+//             flowPane.getChildren().add(item);
+//         }
+
+//         Scene scene = new Scene(flowPane, 400, 400);
+//         stage.setScene(scene);
+//         stage.show();
+//     }
+
+//     public static void main(String[] args) {
+//         launch(args);
+//     }
+// }
\ No newline at end of file
Index: src/main/java/edu/sjsu/cs/User/Seller.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/edu/sjsu/cs/User/Seller.java b/src/main/java/edu/sjsu/cs/User/Seller.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/edu/sjsu/cs/User/Seller.java	
@@ -0,0 +1,47 @@
+package edu.sjsu.cs.User;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Seller extends User {
+    private String companyName;
+    private String companyAddress;
+    public List<Product> sellerItems;
+
+    public Seller(String username, String password, String email, String phone, String companyName, String companyAddress) {
+        super(username, password, email, phone);
+        this.companyName = companyName;
+        this.companyAddress = companyAddress;
+        sellerItems = new ArrayList<>();
+    }
+
+    public String getCompanyName() {
+        return companyName;
+    }
+
+    public void setCompanyName(String companyName) {
+        this.companyName = companyName;
+    }
+
+    public String getCompanyAddress() {
+        return companyAddress;
+    }
+
+    public void setCompanyAddress(String companyAddress) {
+        this.companyAddress = companyAddress;
+    }
+
+    public List<Product> getSellerItems(){return sellerItems;}
+
+    public void setSellerItems(List<Product> sellerItems){this.sellerItems = sellerItems; };
+
+
+
+
+    @Override
+    public void viewProfile() {
+        //TODO
+
+    }
+
+}
